Release Godot 4.1.2 and publish release notes (#724)
godotengine/godot-website: collections/_article/maintenance-release-godot-4-1-2.md

---
title: "Maintenance release: Godot 4.1.2"
excerpt: "As Godot 4.2 enters feature freeze this week, a new Godot 4.1 maintenance release arrives with a handful of improvements to stability and performance!"
categories: ["release"]
author: Yuri Sizov
image: /storage/blog/covers/maintenance-release-godot-4-1-2.webp
image_caption_title: Robot Detour
image_caption_description: A game by Nozomu Games
date: 2023-10-04 14:00:00
---

> Godot 4.1 is celebrating its [3 month anniversary](/article/godot-4-1-is-here) this week, and that means Godot 4.2 is entering the beta testing phase shortly after. Each development cycle is fruitful with bugfixes and other sorts of improvements, and it's an integral part of our development process to identify such changes. If these contributions prove to be safe and stable, we compile a bunch of them into a patch release for the current stable version of the engine.

Reviewer: Just a grammar and style suggestion that simplifies things a bit but gets the same point across. 

```suggestion
Godot 4.1 is celebrating its [3 month anniversary](/article/godot-4-1-is-here) this week, and that means Godot 4.2 is entering the beta testing phase shortly after. This development cycle has been packed with bugfixes and other  improvements. As always, we take the safest of these contributions and we compile them into a patch release for the current stable version of the engine.
```

Author: Thanks, that's much better!

Such is the case of Godot 4.1.2, which is officially available starting today! Engine contributors provided fixes for almost every area of the engine, so make sure to read the changelog [below](#changes). Here's a short summary of some of the more interesting changes.

> - A number of performance issues have been addressed in rendering ([GH-80070](https://github.com/godotengine/godot/pull/80070), [GH-80407](https://github.com/godotengine/godot/pull/80407)), GUI ([GH-79325](https://github.com/godotengine/godot/pull/79325), [GH-80857](https://github.com/godotengine/godot/pull/80857)), and engine core ([GH-81037](https://github.com/godotengine/godot/pull/81037)). The latter fix should also improve memory profile of the engine if you instance a lot of plain objects.

Reviewer 2: ```suggestion
- A number of performance issues have been addressed in rendering ([GH-80070](https://github.com/godotengine/godot/pull/80070), [GH-80407](https://github.com/godotengine/godot/pull/80407)), GUI ([GH-79325](https://github.com/godotengine/godot/pull/79325), [GH-80857](https://github.com/godotengine/godot/pull/80857)), and engine core ([GH-81037](https://github.com/godotengine/godot/pull/81037)). The latter fix should also improve the memory profile of the engine if you instance a lot of plain objects.
```

- Another series of changes in rendering squashes a variety of conditions for crashes, and improves the visuals by fixing glitches ([GH-80485](https://github.com/godotengine/godot/pull/80485), [GH-79660](https://github.com/godotengine/godot/pull/79660)) and removing leaks when using the lightmapper on Vulkan ([GH-81872](https://github.com/godotengine/godot/pull/81872)).

- A previously reported issue with Steam Input and gamepads causing some input events to be doubled is addressed by 4.1.2 ([GH-76045](https://github.com/godotengine/godot/pull/76045)). This issue affects the Linux platform, which means it's a very welcome fix for all Steam Deck developers and users.

- Another critical platform-specific bug has been identified for the web platform, with file system operations failing and causing issues for developers and users of Godot's own online editor. It is now fixed ([GH-79866](https://github.com/godotengine/godot/pull/79866)), which means you can once again save your projects while working on the web.

> - As many game developers compile Godot and custom forks by themselves, it's important for us to ensure buildsystem stability and flexibility as well as that of the rest of the engine — even for previous releases. This version includes a series of platform fixes, targeting XCode 15 ([GH-82458](https://github.com/godotengine/godot/pull/82458)), Visual Studio 2022 for macOS ([GH-81802](https://github.com/godotengine/godot/pull/81802)), as well as general Visual Studio and MSVC improvements ([GH-79238](https://github.com/godotengine/godot/pull/79238)).

Reviewer 3: Technically, the VS for Mac change is not for compiling Godot, but for using it as external IDE to edit and compile C# scripts. But I don't think it a big deal to keep it as is either.

Author: Ah, so it shouldn't have been marked as buildsystem then.

- Rendering: GLES3: Fix memory access error for `MultiMesh` ([GH-80788](https://github.com/godotengine/godot/pull/80788)).
- Rendering: Vulkan: Fix texture update ([GH-80781](https://github.com/godotengine/godot/pull/80781)).
- Rendering: Vulkan: Fix crash with many Omni/SpotLights, Decals or ReflectionProbes ([GH-80845](https://github.com/godotengine/godot/pull/80845)).
- Shaders: Allow more hint types for uniform arrays ([GH-79100](https://github.com/godotengine/godot/pull/79100)).
- Shaders: Fix shader type detection ([GH-79287](https://github.com/godotengine/godot/pull/79287)).
- Shaders: Fix Shader and ShaderInclude resource loading ([GH-80705](https://github.com/godotengine/godot/pull/80705)).
- XR: Fix issue with accessing hand tracking without timing info ([GH-78817](https://github.com/godotengine/godot/pull/78817)).
- XR: Ensure OpenXR classes are declared properly ([GH-81037](https://github.com/godotengine/godot/pull/81037)).
- Thirdparty: FreeType 2.13.2, ICU4C 73.2, libpng 1.6.40, libwebp 1.3.2, mbedtls 2.28.4, miniupnpc 2.2.5, openxr 1.0.28, tinyexr 1.0.7.

> - As well as many improvements to the documentation.

Reviewer 3: How does it render with a separation line? Two lists?

Author: No, it's still one list. Just helps visually when editing.
